assets/javascript/apps/pipeline/nodes/widgets.tsx

```diff
@@ -845,9 +849,40 @@ export function HistoryTypeWidget(props: WidgetParams) {
       <div className="flex flex-col">
         <small className="text-red-500">{historyNameError}</small>
       </div>
+
+<div className="flex join">
+      <InputField label="History Mode" help_text="">
+        <select
+          className="select select-bordered join-item"
+          name="history_mode"
+          onChange={(e) => {
+            setHistoryMode(e.target.value);
+            props.updateParamValue(e);
+          }}
+          value={historyMode}
+        >
+          <option value="Summarize">Summarize</option>
+          <option value="Truncate Tokens">Truncate Tokens</option>
+          <option value="Max History Length">Max History Length</option>
+        </select>
+      </InputField>
+    </div>
+
+    {(historyMode === "Summarize" || historyMode === "Truncate Tokens") && (
+      <div className="flex join">
+        <InputField label="Max Token Size" help_text="">
```

Comment:
nit: update label
```suggestion
        <InputField label="Token Limit" help_text="">
```

We should also find a way to add some help text for this and the mode select:

> When this limit is reached, the history will be compressed using the method selected.